### ISSUE TITLE
Height, Width and Padding are going nutz!

### ISSUE THREAD

**Issue by vbenavidesa** [swimlane/ngx-charts#514]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here
```

So ngx-charts is great to use most of the time. my request goes to get a variable to set the height and width independently, I am having a lot of issues on getting the tables to display properly with the automatic solution you have, as my container is dynamic on height it goes to 0 when I insert the chart only, so I have to manually set a height which then goes nutz with the charts axis labels. Another big issues is padding, I saw #225 but if I put another div inside my html it makes the chart HUGE! and doesnt work as a workaround, I use INSPINA dashboard SPA which has some small white boxes called ibox that holds stuff like widgets but the chart goes insane inside it.

So basically I would love 2 features:

- independent height and width variables.
- padding option.

**Comment by marjan-georgiev** [comment]
I'm not sure what you mean by padding option? The just simply expands to fill the size of it's parent container. The parent container should have a height already set, and no padding.

**Comment by vbenavidesa** [comment]
Hi marjan,

Widgster actually has no height set and it adapts to the information you place inside. That is why I was wondering if you will implement an independent width and height feature.

On the padding option, the graphic just gets to the very edge of the widget and the legend sticks out just a bit. if you have an email I can provide some screenshots.

**Comment by marjan-georgiev** [comment]
You could try wrapping the chart in a div which does not have any padding. The div will also need to have its height set, otherwise the auto size will not work.

**Comment by alex88** [comment]
In my case, having set `[view]="[100,100]"` doesn't set the chart to be 100x100 but just the container and the graph has quite some padding

![screen shot 2017-12-11 at 11 01 25 am](https://user-images.githubusercontent.com/436888/33848488-c0bb4352-de62-11e7-84cc-a9dbf2612992.png)

is there a way to not have that padding?

**Comment by dezudas** [comment]
@alex88 I had similar issue, I am getting extra space in bar and pie chart which is killing space of my container.

**Comment by iliyaisd** [comment]
+1 to the request, having the same issue. Struggled with Pie Grid and finally just had to copy/paste and redefine it in a separate component, changing the following things: 
- set `margin = [0, 0, 0, 0];` (not customizable) 
- set `dy="0"` to the text in the svg (absence of label is not customizable) 
- remove the part of svg regarding total that I don't need (absence is not customizable). 
It would be nice to have those things customizable with inputs. 
Thank you for a nice library!

**Comment by mnasyrov** [comment]
I'm using a following workaround for ngx-chart v7.4 that sets chart's margins in `ngAfterViewInit`:

```typescript
import {PieChartComponent} from '@swimlane/ngx-charts';

@Component()
export class MyPage {
    @ViewChild(PieChartComponent) pieChart: PieChartComponent;

    ngAfterViewInit() {
        setTimeout(() => {
            this.pieChart.margin = [0, 0, 0, 0];
            this.pieChart.update();
        }, 0);
    }
}
```

**Comment by soothran** [comment]
I got this working by increasing timout

**Comment by mnasyrov** [comment]
I made another workaround by using a directive which can do the same without timeouts.

```typescript
import {Directive, Self} from '@angular/core';
import {PieChartComponent} from '@swimlane/ngx-charts';

@Directive({
    selector: 'ngx-charts-pie-chart[hotfix-zero-margin]'
})
export class NgxPieChartZeroMarginDirective {
    constructor(@Self() pieChart: PieChartComponent) {
        pieChart.margin = [0, 0, 0, 0];
    }
}
```

**Comment by furevor** [comment]
> I made another workaround by using a directive which can do the same without timeouts.
> 
> ```ts
> import {Directive, Self} from '@angular/core';
> import {PieChartComponent} from '@swimlane/ngx-charts';
> 
> @Directive({
>     selector: 'ngx-charts-pie-chart[hotfix-zero-margin]'
> })
> export class NgxPieChartZeroMarginDirective {
>     constructor(@Self() pieChart: PieChartComponent) {
>         pieChart.margin = [0, 0, 0, 0];
>     }
> }
> ```

Hello! Thank you for this workaround, but it looks like that now it doesn't work. Maybe any other ideas?

**Comment by AaronLavers** [comment]
> I made another workaround by using a directive which can do the same without timeouts.
> 
> ```ts
> import {Directive, Self} from '@angular/core';
> import {PieChartComponent} from '@swimlane/ngx-charts';
> 
> @Directive({
>     selector: 'ngx-charts-pie-chart[hotfix-zero-margin]'
> })
> export class NgxPieChartZeroMarginDirective {
>     constructor(@Self() pieChart: PieChartComponent) {
>         pieChart.margin = [0, 0, 0, 0];
>     }
> }
> ```

I've just tried to implement this directive in ngx-charts 12.0.1 / angular 8, and for anyone else trying you need to make a small change to 'margin', instead making it 'margins'

`pieChart.margins = [0, 0, 0, 0];`

This seems to work quite well. It would be really good if the Swimlane team could implement the margins as an input however.
